src/content/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/opentelemetry-tutorials/opentelemetry-tutorial-java.mdx

```diff
@@ -0,0 +1,1132 @@
+
+            // Set the span as the current span
+            try (var scope = span.makeCurrent()) {
+                if (n < 1 || n > 90) {
+                    throw new IllegalArgumentException("n must be 1 <= n <= 90.");
+                }
+                // Base cases
+                if (n == 1) {
+                    span.setAttribute(ATTR_RESULT, 1);
+                    return 1;
+                }
+                if (n == 2) {
+                    span.setAttribute(ATTR_RESULT, 1);
+                    return 1;
+                }
+
+                long lastLast = 1;
+                long last = 2;
+                for (long i = 4; i <= n; i++) {
+                    long cur = last + lastLast;
+                    lastLast = last;
+                    last = cur;
+                }
+                span.setAttribute(ATTR_RESULT, last);
+                // Counter to capture valid inputs and the number of times each occurs
+                myCounter.add(1, Attributes.of(ATTR_VALID_N, true));
+                return last;
+            } catch (IllegalArgumentException e) {
+                // Record the exception and set the span status
+                span.recordException(e).setStatus(StatusCode.ERROR, e.getMessage());
+                // Counter to capture invalid inputs and the number of times each occurs
+                myCounter.add(1, Attributes.of(ATTR_VALID_N, false));
+                throw e;
+            } finally {
+                // End the span
+                span.end();
+            }
+        }
+        ```
+
+### K. Custom log instrumentation [#cust-log]
+
+The status of the logs signal in OpenTelemetry Java is currently [experimental](https://opentelemetry.io/docs/instrumentation/java/#status-and-releases). Log messages are managed by the root handler in your app, which will send logs of `INFO` level and higher to the console by default. However, you can modify the logger behavior by changing the logging level, including for specific classes, or installing a custom handler or filter.
+
+#### Initialize the logger [#initialize-logger]
+
+Initialize the logger (as previously stated, this is from the `java.util.logging` library. The Logger is not an OpenTelemetry component, but the application has been configured to send Log4j logs to the OpenTelemetry Log SDK):
+
+```java lineHighlight=4-5
+@RestController
+public class Controller {
+
+  // Logger (note that this is not an OTel component)
+  private static final Logger LOGGER = LogManager.getLogger(Controller.class);
+
+  // Attribute constants
+  private static final AttributeKey<Long> ATTR_N = AttributeKey.longKey("fibonacci.n");
+  private static final AttributeKey<Long> ATTR_RESULT = AttributeKey.longKey("fibonacci.result");
+  private static final AttributeKey<Boolean> ATTR_VALID_N = AttributeKey.booleanKey("fibonacci.valid.n");
+. . .
+}
+```
+
+#### Add custom log messages [#cust-log-messages]
+
+You can now use the logger to record:
```

Comment:
the 'now' throws me a bit here, not sure if it's a new feature or if it's directing me to do that step next.

Reply:
@zuluecho9 I modified it to this: Once you've initialized the logger, you can use the logger to record: